rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -50,6 +52,9 @@ SequentialReader::open(
     return;
   }
 
+  const auto bag_metadata = storage_->get_metadata();
+  file_paths_ = bag_metadata.relative_file_paths;
```

Comment:
do we need this? Could we not just use the bag_metadata and store this as a class member?

Reply:
Just in the constructor. Simplified per suggestion above.